### PR TITLE
Fix #357: Scrollbar margin inconsistency

### LINKED ISSUE #357: Scrollbar margin inconsistency
The scrollbar has a different margin in the hovering state which results in different dimensions and jumping containers. Also the roundness is lost and the border is too thick IMHO during hovering.

Example from gtk3-widget-factory:

scrollbar

### PR DESCRIPTION
**Pull request by SammysHP**
Completely refactored scrollbar and added proper stepper support (although they are disabled by default).

Examples with and without steppers:

scrollbar_steppers scrollbar

### CODE REVIEW

File: gtk-3.20/scss/widgets/_scrollbar.scss [hunk 1]
```diff
@@ -27,135 +29,108 @@
...
+                    border-right: 1px solid $borders_color;
+                }
             }
         }
     }
-
-    treeview ~ scrollbar.vertical {
-        border-top: 1px solid $borders_color;
-        margin-top: -1px;
-    }
```
Comment (khurshid-alam) [inline gtk-3.20/scss/widgets/_scrollbar.scss:160]: Why is this removed?

**Comment by khurshid-alam** [comment]
Steppers are obsolete for overlay scrollbar. We should go without it.

File: gtk-3.20/scss/widgets/_scrollbar.scss (hunk 1 above)
Comment (SammysHP) [inline gtk-3.20/scss/widgets/_scrollbar.scss:160]: It added a border where no border is necessary (because there is already a border from the treeview).

**Comment by SammysHP** [comment]
By default they are disabled. But they can be enabled via a custom gtk.css in .config/gtk3.0 and without this code there would be only gray boxes and an ugly scrollbar. Thus I thought it would be wise to keep some code as fallback.

**Comment by SammysHP** [comment]
Shall I fix the style issues? (Although I think that single line declarations are absolutely fine for a single element.)

**Comment by megax** [comment]
@SammysHP Minimal design correction :) :
$_slider_min_length: 40px; (increased usability)
Now (20 px):
image

New:
image

$_slider_small_width: 4px; (Old Numix (circa size))

**Comment by SammysHP** [comment]
@megax 
Will do it in ~5h.

**Comment by SammysHP** [comment]
Applied changes suggested by @megax, fixed some of the doggy issues (but still kept single declarations in one line because I'm of the opinion that this way is much more readable) and rebased on current master.

**Comment by khurshid-alam** [comment]
@SammysHP @megax Did any of you create a pull for fixing terminal (vte) scrollbar before?

**Comment by SammysHP** [comment]
@khurshid-alam See https://github.com/numixproject/numix-gtk-theme/issues/359#issuecomment-213799629.

File: gtk-3.20/scss/widgets/_scrollbar.scss [hunk 2]
```diff
@@ -27,135 +29,106 @@
...
 
                 slider {
-                    margin: 0;
-                    min-width: 5px;
-                    min-height: 5px;
+                    min-width: $_slider_small_width;
+                    min-height: $_slider_small_width;
                     background-color: $fg_color;
-                    border: 1px solid if($variant == 'light', white, black);
+                    border: 1px solid if($variant == "light", white, black);
```
Comment (megax) [inline gtk-3.20/scss/widgets/_scrollbar.scss:65]: No convert `$variant == 'light'` to `$variant == "light"`
white and black convert $balck, $white :)

Comment (SammysHP) [inline gtk-3.20/scss/widgets/_scrollbar.scss:65]: Why single quotes instead of double quotes? The dog asked me to change single quotes into double quotes. :(

I'll fix the colors tomorrow.

Comment (megax) [inline gtk-3.20/scss/widgets/_scrollbar.scss:65]: No bot comment =/ https://github.com/numixproject/numix-gtk-theme/pull/394/files#diff-5b92b8ba7bbabb66d6af49b3d5e3f220R70

Comment (SammysHP) [inline gtk-3.20/scss/widgets/_scrollbar.scss:65]: As you already fixed it in another pull request, this will/can be done during merge.
